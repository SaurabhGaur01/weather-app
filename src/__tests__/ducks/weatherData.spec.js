import {
    setWeatherData,
    clearWeatherData,
    weatherDataReducer,
} from '../../ducks/weatherData';

const mockObject = {
        mockVariable: 'mockValue',
};

describe('weatherData Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setWeatherData should match the snapshot' , () => {
            expect(setWeatherData(mockObject)).toMatchSnapshot();
        });
        it('clearWeatherData should match the snapshot' , () => {
            expect(clearWeatherData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(weatherDataReducer()).toEqual({});
        });
        it('should return the passed state if called with no action' , () => {
            expect(weatherDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(weatherDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of the API call' , () => {
            expect(weatherDataReducer('', setWeatherData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearWeatherData' , () => {
            expect(weatherDataReducer(mockObject, clearWeatherData())).toEqual({});
        });
    });
});