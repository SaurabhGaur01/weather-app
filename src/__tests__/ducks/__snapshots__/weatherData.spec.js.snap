// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weatherData Duck Tests Action Generator clearWeatherData should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/CLEAR_WEATHER_DATA",
}
`;

exports[`weatherData Duck Tests Action Generator setWeatherData should match the snapshot 1`] = `
Object {
  "data": Object {
    "mockVariable": "mockValue",
  },
  "type": "modules/weather-app/SET_WEATHER_DATA",
}
`;
